core/src/main/scala/kafka/server/metadata/DynamicClientQuotaPublisher.scala

```diff
@@ -48,9 +51,20 @@ class DynamicClientQuotaPublisher(
   ): Unit = {
     val deltaName = s"MetadataDelta up to ${newImage.highestOffsetAndEpoch().offset}"
     try {
-        Option(delta.clientQuotasDelta()).foreach { clientQuotasDelta =>
-          clientQuotaMetadataManager.update(clientQuotasDelta)
+      quotaManagers.clientQuotaCallback().ifPresent(clientQuotaCallback => {
+        if (delta.topicsDelta() != null || delta.clusterDelta() != null) {
```

Comment:
How about moving this callback trigger to a individual publisher? That can decouple the callback and other publisher. Also, that can avoid producing incorrect error message?